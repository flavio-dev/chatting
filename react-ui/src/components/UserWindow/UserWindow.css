.WindowWrapper {
  flex: 1 0 200px; }
  @media only screen and (min-width: 601px) {
    .WindowWrapper {
      flex: 1 0 400px;
      margin: 15px; } }

.WindowTitle {
  align-items: center;
  color: #3A2424;
  display: flex;
  font-family: "Lobster", Helvetica, sans-serif;
  font-size: 30px;
  text-transform: capitalize;
  margin-left: 10px; }

.WindowOnline {
  background-color: #95FF87;
  border-radius: 10px;
  margin-right: 10px;
  height: 15px;
  width: 15px; }

.Window {
  border: 5px solid #3A2424;
  border-radius: 5px;
  color: #0f1121; }

.WindowContent {
  margin-left: auto;
  margin-right: auto;
  width: 95%;
  height: 300px;
  padding: 20px 0; }
  @media only screen and (min-width: 601px) {
    .WindowContent {
      width: 90%; } }

.WindowYou, .WindowThem {
  align-items: flex-end;
  border-bottom: 1px dotted #00A0CC;
  display: flex; }
  .WindowYou span, .WindowThem span {
    margin-right: 5px;
    text-transform: capitalize; }

.WindowYou {
  justify-content: flex-end;
  padding: 10px 0 10px 30%; }
  .WindowYou span {
    color: #3A2424; }

.WindowThem {
  padding: 10px 30% 10px 0; }
  .WindowThem span {
    color: #B78686; }

.WindowCTA {
  margin-left: auto;
  margin-right: auto;
  width: 95%;
  overflow: hidden; }
  @media only screen and (min-width: 601px) {
    .WindowCTA {
      width: 90%; } }
  .WindowCTA textarea {
    border: 2px solid #3A2424;
    border-radius: 5px;
    height: 36;
    margin-bottom: 20px;
    padding: 3%;
    width: 92%;
    max-width: 92%; }
    @media only screen and (min-width: 601px) {
      .WindowCTA textarea {
        width: 93%;
        max-width: 93%; } }
  .WindowCTA button {
    transition: all 0.3s ease;
    background-color: #00A0CC;
    border: 5px solid #00A0CC;
    border-radius: 5px;
    color: #F4F9FF;
    float: right;
    font-family: "Lobster", Helvetica, sans-serif;
    font-size: 20px;
    margin-bottom: 20px;
    padding: 5px 15px; }
    .WindowCTA button:hover {
      background-color: #F4F9FF;
      color: #00A0CC; }
