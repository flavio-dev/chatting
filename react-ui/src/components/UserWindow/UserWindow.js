import React, { Component } from 'react'
import PropTypes from 'prop-types'

import styles from './UserWindow.css'

class UserWindow extends Component {
  constructor(props) {
    super(props)
    this.state = {
      textMessage: ''
    }
    this.typing = this.typing.bind(this)
    this.sendMessage = this.sendMessage.bind(this)
  }

  typing(event) {
    this.setState(
      {textMessage: event.target.value}
    )
  }

  sendMessage() {
    this.props.sendMessage(this.state.textMessage, this.props.you, this.props.to)
    this.setState(
      {textMessage: ''}
    )
  }

  render() {
    return (
      <div className={styles.WindowWrapper}>
        <h2 className={styles.WindowTitle}>{this.props.online &&
          <div className={styles.WindowOnline} />
        }&nbsp;{this.props.title}</h2>
        <div className={styles.Window}>
          <div className={styles.WindowContentWrapper}>
            <div className={styles.WindowContent}>
              {this.props.messages.map((message, index) => {
                if (message.from === this.props.you) {
                  return (
                    <div className={styles.WindowYou} key={index}>
                      <span>you:&nbsp;</span>{message.message}
                    </div>
                  )
                } else {
                  return (
                    <div className={styles.WindowThem} key={index}>
                      <span>{message.from}:&nbsp;</span>{message.message}
                    </div>
                  )
                }
              })}
            </div>
          </div>
          <div className={styles.WindowCTA}>
            <textarea value={this.state.textMessage} onChange={this.typing} placeholder='Type a message...' />
            <button onClick={this.sendMessage}>Send</button>
          </div>
        </div>
      </div>
    )
  }
}

UserWindow.propTypes = {
  title: PropTypes.string,
  to: PropTypes.string,
  you: PropTypes.string,
  messages: PropTypes.array,
  online: PropTypes.bool,
  sendMessage: PropTypes.func
}
export default UserWindow
