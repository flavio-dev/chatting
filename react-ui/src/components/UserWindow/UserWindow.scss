@import "globals.scss";

.WindowWrapper {
  flex: 1 0 200px;

  @media #{$medium-and-up} {
    flex: 1 0 400px;
    margin: 15px;
  }
}

.WindowTitle {
  align-items: center;
  color: $color-chocolate;
  display: flex;
  font-family: $title-font-family;
  font-size: 30px;
  text-transform: capitalize;
  margin-left: 10px;
}

.WindowOnline {
  @include online();
}

.Window {
  border: 5px solid $color-chocolate;
  border-radius: 5px;
  color: $color-quiet-black;
}

.WindowContent {
  @include default-wrapper-width();
  height: 300px;
  padding: 20px 0;
}

.WindowYou, .WindowThem {
  border-bottom: 1px dotted $color-blue-sky;
  display: flex;

  span {
    text-transform: capitalize;
  }
}

.WindowYou {
  justify-content: flex-end;
  padding: 10px 0 10px 30%;

  span {
    color: $color-chocolate;
  }
}

.WindowThem {
  padding: 10px 30% 10px 0;

  span {
    color: $color-milk-chocolate;
  }
}

.WindowCTA {
  @include default-wrapper-width();
  overflow: hidden;

  textarea {
    border: 2px solid $color-chocolate;
    border-radius: 5px;
    height: 36;
    margin-bottom: 20px;
    padding: 3%;
    width: 92%;
    max-width: 92%;

    @media #{$medium-and-up} {
      width: 93%;
      max-width: 93%;
    }
  }

  button {
    @include transitions(all);
    background-color: $color-blue-sky;
    border: 5px solid $color-blue-sky;
    border-radius: 5px;
    color: $color-white;
    float: right;
    font-family: $title-font-family;
    font-size: 20px;
    margin-bottom: 20px;
    padding: 5px 15px;

    &:hover {
      background-color: $color-white;
      color: $color-blue-sky;
    }
  }
}
