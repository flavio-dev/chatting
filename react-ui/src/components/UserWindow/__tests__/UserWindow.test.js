'use strict'

import React from 'react'
import UserWindow from '../UserWindow'
import renderer from 'react-test-renderer'

describe('Testing UserWindow.js', () => {
  it('renders correctly with no message', () => {
    const messages = []
    const tree = renderer
      .create(<UserWindow
        title='title'
        to='someone'
        you='me'
        messages={messages}
      />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with a message from me', () => {
    const messages = [
      {
        from: 'me',
        to: 'someone',
        message: 'the message'
      }
    ]
    const tree = renderer
      .create(<UserWindow
        title='title'
        to='someone'
        you='me'
        messages={messages}
      />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with a message from someone', () => {
    const messages = [
      {
        from: 'someone',
        to: 'me',
        message: 'the message two'
      }
    ]
    const tree = renderer
      .create(<UserWindow
        title='title'
        to='someone'
        you='me'
        messages={messages}
      />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
