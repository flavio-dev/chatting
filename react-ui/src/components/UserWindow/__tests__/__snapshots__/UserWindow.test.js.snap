// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UserWindow.js renders correctly with a message from me 1`] = `
<div
  className="WindowWrapper"
>
  <h2
    className="WindowTitle"
  >
     
    title
  </h2>
  <div
    className="Window"
  >
    <div
      className="WindowContentWrapper"
    >
      <div
        className="WindowContent"
      >
        <div
          className="WindowYou"
        >
          <span>
            you: 
          </span>
          the message
        </div>
      </div>
    </div>
    <div
      className="WindowCTA"
    >
      <textarea
        onChange={[Function]}
        placeholder="Type a message..."
        value=""
      />
      <button
        onClick={[Function]}
      >
        Send
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing UserWindow.js renders correctly with a message from someone 1`] = `
<div
  className="WindowWrapper"
>
  <h2
    className="WindowTitle"
  >
     
    title
  </h2>
  <div
    className="Window"
  >
    <div
      className="WindowContentWrapper"
    >
      <div
        className="WindowContent"
      >
        <div
          className="WindowThem"
        >
          <span>
            someone
            : 
          </span>
          the message two
        </div>
      </div>
    </div>
    <div
      className="WindowCTA"
    >
      <textarea
        onChange={[Function]}
        placeholder="Type a message..."
        value=""
      />
      <button
        onClick={[Function]}
      >
        Send
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing UserWindow.js renders correctly with no message 1`] = `
<div
  className="WindowWrapper"
>
  <h2
    className="WindowTitle"
  >
     
    title
  </h2>
  <div
    className="Window"
  >
    <div
      className="WindowContentWrapper"
    >
      <div
        className="WindowContent"
      />
    </div>
    <div
      className="WindowCTA"
    >
      <textarea
        onChange={[Function]}
        placeholder="Type a message..."
        value=""
      />
      <button
        onClick={[Function]}
      >
        Send
      </button>
    </div>
  </div>
</div>
`;
